product_packaging_level/models/product_packaging.py

```diff
@@ -116,16 +120,17 @@ def _make_qty_per_level(self, qty_per_level_mapping, format_pattern=None):
             res.append((code, new_qty))
         return res
 
-    @api.onchange("packaging_level_id")
-    def _onchange_name(self):
-        if self.packaging_level_id:
-            self.name = self.packaging_level_id.name
+    @api.onchange("package_type_id")
+    def _onchange_package_type(self):
+        package_type_level_id = self.package_type_id.packaging_level_id
+        if package_type_level_id:
+            self.packaging_level_id = package_type_level_id
 
-    def name_get(self):
-        result = []
-        for record in self:
-            if record.product_id and record.packaging_level_id:
-                result.append((record.id, record.packaging_level_id.display_name))
-            else:
-                result.append((record.id, record.name))
-        return result
+    @api.onchange("packaging_level_id", "package_type_id", "name", "name_policy")
+    def _onchange_name(self):
```

Comment:
@anothingguy Why not a computed fields with read_only="False" This change is annoying since any packaging created from the python code is created with the wrong name. Moreover, by replacing the `name_get` by an `on_change` you  broke existing installation since the name is no more properly displayed for existing records.